### PR TITLE
Support SD2 attention slicing

### PR DESCRIPTION
**Pull request by anton-l**
Supports attention slicing for `attention_head_dim` defined as a list of head dims, as in Stable Diffusion 2.
Also adds tests for SD2 text2image.

### CODE REVIEW

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py [hunk 1]
```diff
@@ -165,9 +165,14 @@ def enable_attention_slicing(self, slice_size: Optional[Union[str, int]] = "auto
...
-            # speed and memory
-            slice_size = self.unet.config.attention_head_dim // 2
+            if isinstance(self.unet.config.attention_head_dim, int):
+                # half the attention head size is usually a good trade-off between
+                # speed and memory
+                slice_size = self.unet.config.attention_head_dim // 2
+            else:
+                # if `attention_head_dim` is a list, take the smallest head size
+                slice_size = min(self.unet.config.attention_head_dim)
+
```
Comment (anton-l) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py:175]: The base logic for slice calculation, other pipelines are auto-copies of this

**Comment by anton-l** [comment]
For the reviewers: the only parts that aren't auto-copied are my two PR comments above, and the tests

**Comment by anton-l** [comment]
The added slow tests pass locally now, if used with https://github.com/huggingface/diffusers/pull/1402/

**Review by patrickvonplaten — APPROVED** [review]
Looks great thanks!

**Pull request merged**